tc_aws/loaders/s3_loader.py

```diff
@@ -32,6 +34,11 @@ def load(context, url, callback):
         def handle_data(file_key):
             if not file_key or 'Error' in file_key or 'Body' not in file_key:
                 logger.warn("ERROR retrieving image from S3 {0}: {1}".format(key, str(file_key)))
+                # We will now handle the failure and return a 404.
+                result = LoaderResult()
```

Comment:
I actually would prefer that we return the error that S3 gives us. For example, we might get a 403 access denied. I cant figure out a way to send a custom error message in Thumbor, but Im hoping one of you guys with more experience in Thumbor could contribute here. I dont think its acceptable to just let the loader hang indefinitely.